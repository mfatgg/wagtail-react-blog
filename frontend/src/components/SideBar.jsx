import React from "react";
import TagWidget from "./TagWidget.tsx";
import CategoryWidget from "./CategoryWidget.jsx";
import SearchForm from "./SearchForm.jsx";
import BookCard from "./BookCard.jsx";

function SideBar(props) {
  return (
    <div className="w-full sm:w-1/3 md:w-1/4 lg:w-4/12 px-2">
      <SearchForm {...props} />
      <CategoryWidget {...props} />
      <TagWidget {...props} />
      <BookCard />
    </div>
  );
}

export default SideBar;
