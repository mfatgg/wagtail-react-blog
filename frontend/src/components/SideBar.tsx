import React from "react";
import TagWidget, { TagsList } from "./TagWidget";
import CategoryWidget, { CategoriesList } from "./CategoryWidget";
import SearchForm from "./SearchForm";
import BookCard from "./BookCard";

export type SideBarInterface = CategoriesList & TagsList;

function SideBar({ categoriesList, tagsList }: SideBarInterface) {
  return (
    <div className="w-full sm:w-1/3 md:w-1/4 lg:w-4/12 px-2">
      <SearchForm />
      <CategoryWidget categoriesList={categoriesList} />
      <TagWidget tagsList={tagsList} />
      <BookCard />
    </div>
  );
}

export default SideBar;
