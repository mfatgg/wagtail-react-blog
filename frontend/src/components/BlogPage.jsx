import React from "react";
import TopNav from "./TopNav.jsx";
import Footer from "./Footer.jsx";
import SideBar from "./SideBar.jsx";
import PostPageCardContainer from "./PostPageCardContainer.jsx";

function BlogPage(props) {
  return (
    <div className="flex flex-col min-h-screen">
      <TopNav />

      <div className="w-full max-w-7xl mx-auto px-2 sm:px-6 lg:px-8">
        <div className="flex flex-row flex-wrap py-4">
          <PostPageCardContainer {...props} />
          <SideBar {...props} />
        </div>
      </div>

      <Footer />
    </div>
  );
}

export default BlogPage;
