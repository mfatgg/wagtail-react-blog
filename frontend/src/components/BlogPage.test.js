import React from "react";
import BlogPage from "./BlogPage";
import { screen, render } from "@testing-library/react";
import { blogPageData } from "./mockData";
import { PostPageCardContainer } from "./PostPageCardContainer";
import { SideBar } from "./SideBar";
import camelcaseKeys from "camelcase-keys";
import { MemoryRouter } from "react-router-dom";

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});

jest.mock("./PostPageCardContainer", () => ({
  PostPageCardContainer: jest.fn(({ children }) => (
    <div data-testid="PostPageCardContainer">{children}</div>
  )),
}));

jest.mock("./SideBar", () => ({
  SideBar: jest.fn(({ children }) => (
    <div data-testid="SideBar">{children}</div>
  )),
}));

test("BlogPage Test", () => {
  const data = camelcaseKeys(blogPageData, { deep: true });

  const { asFragment } = render(
    <MemoryRouter>
      <BlogPage {...data} />
    </MemoryRouter>
  );

  expect(screen.queryByTestId("PostPageCardContainer")).toBeInTheDocument();
  expect(PostPageCardContainer).toHaveBeenCalledWith(data, expect.anything());

  expect(screen.queryByTestId("SideBar")).toBeInTheDocument();
  expect(SideBar).toHaveBeenCalledWith(data, expect.anything());

  expect(asFragment()).toMatchSnapshot();
});
