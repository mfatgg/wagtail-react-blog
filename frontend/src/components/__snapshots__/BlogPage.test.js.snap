// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogPage Test 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col min-h-screen"
  >
    <nav
      class="bg-white border-b border-opacity-75 border-gray-300 dark:bg-gray-900 dark:text-white dark:border-gray-500"
      data-headlessui-state=""
    >
      <div
        class="w-full max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
      >
        <div
          class="relative flex items-center justify-between h-16"
        >
          <div
            class="absolute inset-y-0 left-0 flex items-center sm:hidden"
          >
            <button
              aria-expanded="false"
              class="inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-white hover:bg-gray-700 focus:outline-none focus:ring-2 focus:ring-inset focus:ring-white"
              data-headlessui-state=""
              id="headlessui-disclosure-button-:r0:"
              type="button"
            >
              <span
                class="sr-only"
              >
                Open main menu
              </span>
              <svg
                aria-hidden="true"
                class="block h-6 w-6"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </div>
          <div
            class="flex-1 flex items-center justify-center sm:items-stretch sm:justify-start"
          >
            <div
              class="shrink-0 flex items-center"
            >
              <a
                href="/"
              >
                <span
                  class="text-bold text-grey-800"
                >
                  React Wagtail Demo
                </span>
              </a>
            </div>
            <div
              class="hidden sm:block sm:ml-6"
            >
              <div
                class="flex space-x-4"
              >
                <a
                  class="text-gray-500 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="#"
                >
                  Dashboard
                </a>
                <a
                  class="text-gray-500 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="#"
                >
                  Team
                </a>
                <a
                  class="text-gray-500 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="#"
                >
                  Projects
                </a>
                <a
                  class="text-gray-500 hover:bg-gray-700 hover:text-white px-3 py-2 rounded-md text-sm font-medium"
                  href="#"
                >
                  Calendar
                </a>
              </div>
            </div>
          </div>
          <div
            class="absolute inset-y-0 right-0 flex items-center pr-2 sm:static sm:inset-auto sm:ml-6 sm:pr-0"
          >
            <button
              class="bg-gray-800 p-1 rounded-full text-gray-400 hover:text-white focus:outline-none"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="h-6 w-6"
                data-slot="icon"
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21.752 15.002A9.72 9.72 0 0 1 18 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 0 0 3 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 0 0 9.002-5.998Z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="w-full max-w-7xl mx-auto px-2 sm:px-6 lg:px-8"
    >
      <div
        class="flex flex-row flex-wrap py-4"
      >
        <div
          data-testid="PostPageCardContainer"
        />
        <div
          data-testid="SideBar"
        />
      </div>
    </div>
    <footer
      class="mt-auto py-7 bg-white border-t border-opacity-75 border-gray-300 dark:bg-gray-900 dark:text-white dark:border-gray-500"
    >
      <div
        class="container mx-auto"
      >
        <p
          class="m-0 text-center text-blue-700 hover:underline dark:text-gray-400"
        >
          Built by 
          <a
            href="https://www.accordbox.com/"
          >
            michaelyin@accordbox.com
          </a>
        </p>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
